appveyor.yml

```diff
@@ -0,0 +1,66 @@
+# shallow clone
+clone_depth: 10
+
+cache:
+  - C:\tools\vcpkg\installed\
+
+os: Visual Studio 2017
+
+environment:
+  # Tell msys2 to add mingw64 to the path
+  MSYSTEM: MINGW64
+  # vcpkg should install 64 bit packages
+  VCPKG_DEFAULT_TRIPLET: x64-windows
+  # Tell msys2 to inherit the current directory when starting the shell
+  CHERE_INVOKING: 1
+  matrix:
+    # - BUILD_TYPE: mingw
+    - BUILD_TYPE: msvc
+
+platform:
+  - x64
+
+configuration:
+  - Release
+
+install:
+  - git submodule update --init --recursive
+  - ps: |
+        if ($env:BUILD_TYPE -eq 'mingw') {
+          # redirect err to null to prevent warnings from becoming errors
+          C:\msys64\usr\bin\bash -lc "pacman --noconfirm -S mingw-w64-x86_64-SDL2 mingw-w64-x86_64-glew mingw-w64-x86_64-devil mingw-w64-x86_64-openal 2>&1"
+          # install unicorn from source
+          mkdir mingw-w64-unicorn
+          cd mingw-w64-unicorn
```

Comment:
as the build folder name should entirely be based on current platform and current compiler, not hardcoded, this can be a more generic folder like "unicorn"